ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/multiplechoice/MultipleChoiceTaskViewModel.kt

```diff
@@ -40,10 +41,14 @@ class MultipleChoiceTaskViewModel @Inject constructor(resources: Resources) :
   private val selectedIds: MutableSet<String> = mutableSetOf()
   private var otherText: String = ""
 
-  val textWatcher =
+  val otherTextWatcher =
     object : TextWatcher {
       override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {
         otherText = s.toString()
+        // Set the other option.
+        _items.value
+          .firstOrNull { it.isOtherOption }
+          ?.let { setItem(it, otherText != "", it.cardinality == SELECT_MULTIPLE) }
```

Comment:
Do we also want to select it when its a radio button and the user start input?

If so we can remove the cardinality check here and in the object itself and just always add the otherId to the selected list on text change.

Reply:
Should instead we make the "other" text required? If we follow the model that Forms uses, it's required if the field is required, optional if not.  In the latter case, we empty other text is a valid state, so we wouldn't want to deselect.

Reply:
> Do we also want to select it when its a radio button and the user start input?

Ah, yes! The implementation works for both types. In this case, we're setting the `canSelectMultiple` parameter based on the cardinality of the input.

![Screenshot 2024-03-15 at 2 44 02 PM](https://github.com/google/ground-android/assets/12646706/e346b1f4-ed39-43dd-9afa-d5fa08c485d7)

> If the text is empty here i.e. the user deleted everything do we also want to deselect?
> In the latter case, we empty other text is a valid state, so we wouldn't want to deselect.
Thank you for raising this, I will definitely refactor to consider the `isRequired` state for the task.